protocol/chainlib/rest.go

```diff
@@ -204,7 +204,9 @@ func (apil *RestChainListener) Serve(ctx context.Context) {
 		utils.LavaFormatInfo("in <<<", &map[string]string{"path": path, "dappID": dappID, "msgSeed": msgSeed})
 		requestBody := string(c.Body())
 		reply, _, err := apil.relaySender.SendRelay(ctx, path, requestBody, http.MethodPost, dappID, metricsData)
-		go apil.logger.AddMetric(metricsData, err != nil)
+		if apil.logger.ShouldCountMetricForHttp(c) {
```

Comment:
for each one of the should please check the env if nil or not other wise we don't need to run it (another wrapper inside)